### ISSUE TITLE
Warning to save work before closing event

### ISSUE THREAD

**Issue by saumier** [culturecreates/footlight-app#75]
When editing an event, before leaving the screen to avoid losing data, a notification appears asking the user to save their work.  

We are looking for the simplest version. A simple version of this can be "You have unsaved changes. Press cancel to continue editing." with the buttons cancel and Ok. Clicking cancel keeps you on the current page. Clicking Ok leaves the page and you lose your unsaved changes.

Copy for this warning is available on tab 2 in https://docs.google.com/spreadsheets/d/1Wh2a-ZHVs94L7JPrgxv-sWRFm6_m7tfXSdPIln0GZiI/edit#gid=2114823229 

## Test Cases

1. Create a new event and start entering data
2. Click "events" in the left nav --> warning to save work should appear
3. Click user menu in top right and then logout --> warning to save work should appear
4. Click the browser back arrow to return to the list of events --> warning to save work should appear

**Comment by AbhishekPAnil** [comment]
ESTIMATE 4hrs

**Comment by troughc** [comment]
Tested in production on chrome.
1, 2, and 3 are fine.
4. click the browser bakc arrow, I don't get a warning.

One other issue that surfaced ( mentioned in slack channel related to the publish button):
When an event is published and I use the dropdown and select published (because I changed my mind), it sets the event to draft (not what I want - it should stay published). Then when I leave I get a warning that my changes aren't saved - I don’t save my changes, but it sets the event to draft anyway (changes it from Published).
